### ISSUE TITLE
Some notes on Order Risks

### ISSUE THREAD

**Issue by MalikDrako** [nozzlegear/ShopifySharp#71]
I have been working on a program that needs to check Order Risks, here are some notes from my partial implementation and questions to Shopify support:

I copied from another entity and service, I think it was Transaction. I made minimal modifications to add the different fields, and only implemented the get methods.

Order Risks contain an undocumented field, "merchant_message". In some cases, but not all, it is a duplicate of the "message" field. Support initially claimed the field was deprecated since it has been undocumented and is a duplicate of "message". When I found ones that were not duplicates and pointed it out to support, they reported that a Solutions Engineer confirmed that "message" is what Shopify sees in logs and "merchant_message" is what the merchant sees on the order. In a manual test, it ignored "merchant_message" and set both to the value I gave in "message". I was not able to get it to set a unique value for "merchant_message".

```
{ /* sent */
    "risk": {
        "message": "Test Med risk flag (message)",
        "merchant_message": "Test Med risk flag (merchant_message)",
        "recommendation": "investigate",
        "score": 0.5,
        "source": "External",
        "cause_cancel": false,
        "display": true
    }
}
```

```
{ /* received */
    "risk": {
        "id": <risk ID>
        "order_id": <order ID>
        "checkout_id": <checkout ID>
        "source": "External"
        "score": "0.5"
        "recommendation": "investigate"
        "display": true
        "cause_cancel": false
        "message": "Test Med risk flag (message)"
        "merchant_message": "Test Med risk flag (message)"
    }
}
```

**Comment by nozzlegear** [comment]
Thanks for reporting this! I've run into the same problem in a few other objects where there are fields that are undocumented but seem to have useful values. I prefer to leave out undocumented properties, since Shopify could remove them at any time. Do you think leaving `merchant_message` out would be safe, or do you think it would break your app?

**Comment by MalikDrako** [comment]
I originally left it out, then put it in when I got an error - you would at least need to account for the field when deserializing.

There are a few cases where the values are different, and I dont think I've seen the value from `message` in the admin, just the value from `merchant_message` when they are different. One option would be to add a property that returns `merchant_message` if it is not null, `message` otherwise.

```
"message": "The Minfraud score (99.0) was above the threshold."
"merchant_message": "There is a high risk of this order being fraudulent. Shopify's fraud analysis has detected details that appear suspicious. Contact the customer to validate the order."
```
```
"message": "Customer was using a high risk web proxy to connect to the internet."
"merchant_message": "The customer used a high risk Internet connection (web proxy) to place this order."
```
```
"message": "Credit card associated with fraudulent transactions."
"merchant_message": "The customer's credit card used for this order has been associated with fraudulent transactions."
```

**Comment by MalikDrako** [comment]
I just noticed you started on this and have some more notes

For `source`, I've seen values "Internal" and "Gateway"

For `merchant_message`, I would add a note that it is not possible to set the value directly using the API, but if something else sets it differently, it is the message shown on the admin, not `message`

**Comment by nozzlegear** [comment]
Thanks! I've added 'Internal' and 'Gateway' to the list of known values, and added your notes on the merchant message.

**Comment by nozzlegear** [comment]
Order Risks are now available in ShopifySharp v3.1.0. Thanks!
